src/celeritas/track/SortTracksAction.cc

```diff
@@ -100,6 +103,26 @@ void SortTracksAction::execute(CoreParams const&, CoreStateHost& state) const
 void SortTracksAction::execute(CoreParams const&, CoreStateDevice& state) const
 {
     detail::sort_tracks(state.ref(), track_order_);
+    state.count_tracks_per_action(track_order_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Set host data at the beginning of a run
+ */
+void SortTracksAction::begin_run(CoreParams const& params, CoreStateHost& state)
+{
+    state.resize_offsets(params.action_reg()->num_actions() + 1);
```

Comment:
I don't think there is a way to resize `thread_offsets`here if it's a member variable of `CoreStateData` as we don't have ownership of it and `resize` functions want value ownership